core/src/main/scala/com/nvidia/spark/rapids/tool/tuning/AutoTuner.scala

```diff
@@ -595,18 +590,17 @@ class AutoTuner(
    *         if version > 3.3.0 recommend "spark.executor.memoryOverheadFactor" and add comment
    *         else recommend "spark.kubernetes.memoryOverheadFactor" and add comment if missing
    */
-  def addRecommendationForMemoryOverhead(recomValue: String): Unit = {
-    if (autoTunerConfigsProvider
-        .enableMemoryOverheadRecommendation(getPropertyValue("spark.master"))) {
-      val memOverheadLookup = memoryOverheadLabel
+  private def addRecommendationForMemoryOverhead(recomValue: String): Unit = {
+    if (!sparkMaster.contains(Standalone)) {
+      val memOverheadLookup = autoTunerConfigsProvider.getMemoryOverheadLabel(sparkMaster,
+        appInfoProvider.getSparkVersion)
+      val pinnedPoolSizeLookup = "spark.rapids.memory.pinnedPool.size"
       appendRecommendationForMemoryMB(memOverheadLookup, recomValue)
-      getPropertyValue("spark.rapids.memory.pinnedPool.size").foreach { lookup =>
-        if (lookup != "spark.executor.memoryOverhead") {
```

Comment:
This comparison was incorrect because `lookup` is a memory string value, whereas `spark.executor.memoryOverhead` is a label